spec/models/async_transaction/base_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe AsyncTransaction::Base, type: :model do
+  describe 'Validation' do
+    let(:transaction1) { create(:address_transaction) }
+    let(:transaction2) { build(:address_transaction) }
+    let(:transaction3) { build(:address_transaction, source: nil) }
+    let(:transaction4) { build(:address_transaction) }
+
+    it 'ensures uniqueness across source and transaction_id' do
+      transaction2.id = transaction1.id
```

Comment:
This is actually testing the `id` attribute, instead of the `transaction_id` attribute.  So you'll want to change this to:

```ruby
transaction2.transaction_id = transaction1.transaction_id
```